indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
             style1.fontSize = modifiedFontSize(18, self._fontsize)
             style1.leading = modifiedFontSize(22, self._fontsize)
             style1.alignment = TA_CENTER
-            self._story.append(Spacer(inch, 1*cm))
-            p = Paragraph( _("Table of contents"), style1)
-            self._story.append(p)
-            self._story.append(Spacer(inch, 1*cm))
-            if len(self._showSessions)>0:
-                style2=ParagraphStyle({})
-                style2.fontSize=modifiedFontSize(14, self._fontsize)
-                style2.leading=modifiedFontSize(10, self._fontsize)
-                style2.alignment=TA_CENTER
-                caption=[]
-                for sessionId in self._showSessions:
-                    session=self._conf.getSessionById(sessionId)
-                    if session is not None:
-                        caption.append(session.getTitle())
-                p=Paragraph("%s"%"\n".join(caption),style2)
+            self._story.append(Spacer(inch, 1 * cm))
+            self._story.append(Paragraph(_("Table of contents"), style1))
+            self._story.append(Spacer(inch, 1 * cm))
+            if self._showSessions:
+                style2 = ParagraphStyle({})
+                style2.fontSize = modifiedFontSize(14, self._fontsize)
+                style2.leading = modifiedFontSize(10, self._fontsize)
+                style2.alignment = TA_CENTER
+                caption = []
+                for session_id in self._showSessions:
+                    sess = self._conf.as_event.sessions.filter_by(id=session_id, is_deleted=False).first()
+                    if sess:
+                        caption.append(sess.title)
```

Comment:
`sessions` is not dynamic anymore. Something like `[sess.title for self._conf.as_event.sessions if (sess.id in self._showSessions and not sess.is_deleted)]` should do the trick (not tested!)